pkg/operator/boundsatokensignercontroller/controller.go

```diff
@@ -0,0 +1,296 @@
+package boundsatokensignercontroller
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"k8s.io/klog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	workQueueKey = "key"
+
+	operatorNamespace = operatorclient.OperatorNamespace
+	targetNamespace   = operatorclient.TargetNamespace
+
+	keySize                  = 2048
+	NextSigningKeySecretName = "next-service-account-signing-key"
+	SigningKeySecretName     = "service-account-signing-key"
+	PrivateKeyKey            = "service-account.key"
+	PublicKeyKey             = "service-account.pub"
+
+	TokenReadyAnnotation = "kube-apiserver.openshift.io/ready-to-use"
+	readyInterval        = 5 * time.Minute
+
+	CertConfigMapName = "bound-sa-token-signing-certs"
+)
+
+// BoundSATokenSignerController manages the keypair used to sign bound
+// tokens and the key bundle used to verify them.
+//
+// This code is derived from the controller used by the
+// kube-controller-manager-operator to manage the key material for the
+// legacy sa token signer.
+type BoundSATokenSignerController struct {
+	secretClient    corev1client.SecretsGetter
+	configMapClient corev1client.ConfigMapsGetter
+	eventRecorder   events.Recorder
+
+	cachesSynced []cache.InformerSynced
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewBoundSATokenSignerController(
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+
```

Comment:
nit: no empty line

Reply:
Done.  I was just being consistent with [the controller this was derived from](https://github.com/openshift/cluster-kube-controller-manager-operator/blob/master/pkg/operator/certrotationcontroller/satokensigner_controller.go#L58), I guess that could use an update too.